### PR TITLE
feat: automated-region-configuration (IDE-732)

### PR DESCRIPTION
**Pull request by acke**
### Description

If $/snyk.hasAuthenticated transmits an API URL, this is saved in the settings.

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [x] CHANGELOG.md updated

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: src/main/kotlin/snyk/common/CustomEndpoints.kt [hunk 1]
```diff
@@ -21,14 +21,12 @@ fun toSnykCodeApiUrl(endpointUrl: String?): String {
         uri.isDev() ->
             endpoint
                 .replace("api.", "")
-                .replace("/v1", "")
```
Comment (acke) [inline src/main/kotlin/snyk/common/CustomEndpoints.kt:24]: dont remove the /v1 check

File: src/main/kotlin/snyk/common/CustomEndpoints.kt [hunk 2]
```diff
@@ -21,14 +21,12 @@ fun toSnykCodeApiUrl(endpointUrl: String?): String {
...
                 .replace("https://dev.", "https://$codeSubdomain.dev.")
                 .suffixIfNot("/")
 
         uri.isSnykTenant() ->
             endpoint
                 .replace("https://api.", "https://")
-                .replace("/v1", "")
```
Comment (acke) [inline src/main/kotlin/snyk/common/CustomEndpoints.kt:31]: dont remove the /v1 check

File: src/main/kotlin/snyk/common/CustomEndpoints.kt (hunk 1 above)
Comment (acke) [reply src/main/kotlin/snyk/common/CustomEndpoints.kt:24]: Done.

File: src/main/kotlin/snyk/common/CustomEndpoints.kt (hunk 2 above)
Comment (acke) [reply src/main/kotlin/snyk/common/CustomEndpoints.kt:31]: Done.

File: src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt [hunk 3]
```diff
@@ -270,8 +270,11 @@ class SnykLanguageClient :
     fun hasAuthenticated(param: HasAuthenticatedParam) {
         if (disposed) return
         val oldToken = pluginSettings().token
-        if (oldToken == param.token) return
+        val oldApiUrl = pluginSettings().customEndpointUrl
+        if (oldToken == param.token && oldApiUrl == param.apiUrl) return
```
Comment (bastiandoetsch) [inline src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt:274]: this should be separated.

File: src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt [hunk 4]
```diff
@@ -270,8 +270,11 @@ class SnykLanguageClient :
...
         pluginSettings().token = param.token
+        pluginSettings().customEndpointUrl = param.apiUrl
```
Comment (bastiandoetsch) [inline src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt:276]: param.apiUrl needs to be checked against null/blank

File: src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt [hunk 5]
```diff
@@ -270,8 +270,11 @@ class SnykLanguageClient :
     fun hasAuthenticated(param: HasAuthenticatedParam) {
         if (disposed) return
         val oldToken = pluginSettings().token
-        if (oldToken == param.token) return
+        val oldApiUrl = pluginSettings().customEndpointUrl
+        if (oldToken == param.token && oldApiUrl == param.apiUrl) return
         pluginSettings().token = param.token
```
Comment (bastiandoetsch) [inline src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt:275]: this should only be overwritten, if it's unequal (empty is allowed)

File: src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt (hunk 4 above)
Comment (bastiandoetsch) [inline src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt:276]: API URL should be updated before the token.

File: src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt (hunk 5 above)
Comment (acke) [reply src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt:280]: Fixed

**Pull request merged**